### PR TITLE
feat. history file improvements

### PR DESCRIPTION
**Pull request by Rendevior**
- minor changes

### CODE REVIEW

**Comment by Rendevior** [comment]
soon will update the preview video. :)
Already fixed all the problems integrating from v2.4.1 -> v2.4.2

**Comment by Rendevior** [comment]
found out that `sed -i` isn't a POSIX, so we're going back to good ol' iterating/slow loops, instead of using shell jobs to iterate simultainiously. though, i'm still gonna improve it a little.

**Comment by Rendevior** [comment]
> found out that `sed -i` isn't a POSIX, so we're going back to good ol' iterating/slow loops, instead of using shell jobs to iterate simultainiously. though, i'm still gonna improve it a little.

Aight, i knew it.. there was a POSIX-way workaround of `sed -i`
I'm done with this one.

File: hayanime [hunk 1]
```diff
@@ -232,8 +232,8 @@ ffmpegdler(){
 hist_handler(){
 	[ -e ~/.hayanihist ] || touch ~/.hayanihist 2>/dev/null		# Supress Incase of the Directory is Unwritable
 	if [ "${1}" = "append" ]; then
-		[ -w ~/.hayanihist ] || die "1" "Hist file is Unwritable, Make sure you have access to write to HOME directory"
-		sed "/${3}/d" ~/.hayanihist > ~/.hayanihist.tmp && mv ~/.hayanihist.tmp ~/.hayanihist
+		[ -w ~/.hayanihist ] || die 1 "Hist file is Unwritable, Make sure you have access to write to HOME directory"
+		printf '%s' "$(sed "/${3}/d" ~/.hayanihist)" > ~/.hayanihist
```
Comment (Rendevior) [inline hayanime:236]: Note that it expands the command substitution first before writing/redirecting it directly to the same file

File: hayanime [hunk 2]
```diff
@@ -332,14 +332,22 @@ hist_handler(){
...
+			tot_ep="$(curl -sLk "${base_url}${entry%%¢*}" -A "${user_agent}" | sed -nE 's_.*eptotal":([^\,\}]*)(,.*|\}</.*)_\1_p')"
+			[ -z "${tot_ep}" ] && printf "\033[38;5;1m\033[2K\r%b\033[0m" "Failed to Retrieve data." && : > .hayanihist.tmp && exit 1
+			printf '%s' "$(sed "${entriter}s%\(.*€\)\(.*\)\(£.*\)%\\1${cur_ep}/${tot_ep}\\3%" ~/.hayanihist.tmp)" > ~/.hayanihist.tmp
+		) &
 		progbar "${entriter}" "${hist_count}" "Updating..."
+		[ "$(jobs -p | awk 'BEGIN{c=0} //{c++} END{print c}')" -gt 2 ] && wait "${!}"
 		unset cur_ep
 	done < ~/.hayanihist
+	wait "${!}"
+	[ -z "$(cat ~/.hayanihist.tmp)" ] && die 1 "\033[2K\rFailed to Update..."
```
Comment (Rendevior) [inline hayanime:354]: If it fails it will not write the update to the main history file

**Comment by fearocanity** [comment]
Zamn, all working fine and already tested it. this version would be my primary one.

**Review by fearocanity — APPROVED** [review]

**Pull request merged**